libflux/src/flux/semantic/flatbuffers/mod.rs

```diff
@@ -1,3 +1,1147 @@
+
+            walk::Node::BuiltinStmt(builtin) => {
+                let id = v.pop_ident("built".to_string());
+                let builtin = fbsemantic::BuiltinStatement::create(
+                    &mut v.builder,
+                    &fbsemantic::BuiltinStatementArgs { loc, id },
+                );
+                v.stmts.push((
+                    builtin.as_union_value(),
+                    fbsemantic::Statement::BuiltinStatement,
+                ));
+            }
+
+            walk::Node::OptionStmt(opt) => {
+                let (assignment, assignment_type) = {
+                    match &opt.assignment {
+                        semantic::nodes::Assignment::Variable(_) => match v.stmts.pop() {
+                            Some((nva, fbsemantic::Statement::NativeVariableAssignment)) => {
+                                (Some(nva), fbsemantic::Assignment::NativeVariableAssignment)
+                            }
+                            Some((_, ty)) => {
+                                v.err =
+                                    Some(String::from(format!("found {:?} in stmt vector", ty)));
+                                return;
+                            }
+                            None => {
+                                v.err = Some(String::from(
+                                    "Native assignment was not added to SerializingVisitor",
+                                ));
+                                return;
+                            }
+                        },
+                        semantic::nodes::Assignment::Member(_) => match v.stmts.pop() {
+                            Some((member, fbsemantic::Statement::MemberAssignment)) => {
+                                (Some(member), fbsemantic::Assignment::MemberAssignment)
+                            }
+                            _ => {
+                                v.err = Some(String::from(
+                                    "Member assignment was not added to SerializingVisitor",
+                                ));
+                                return;
+                            }
+                        },
+                    }
+                };
+                let opt = fbsemantic::OptionStatement::create(
+                    &mut v.builder,
+                    &fbsemantic::OptionStatementArgs {
+                        loc,
+                        assignment,
+                        assignment_type,
+                    },
+                );
+                v.stmts
+                    .push((opt.as_union_value(), fbsemantic::Statement::OptionStatement));
+            }
+
+            walk::Node::Block(block) => {
+                // Block statements must be convered in this enum in order to also walk all child nodes.
+                // However, block statements are consumed by the function expression node so that
+                // blocks are only constructed once all child nodes have been traversed.
+            }
```

Comment:
Should we add an `unreachable!()` here? To ensure we never actually find ourselves in that block?

Reply:
Hmm, I haven't used that warning before. Let me look into it.

Reply:
This arm will be reached (and should be) it's just that it doesn't need to do anything.  Blocks are kind of confusing since each one is essentially a node in a linked list.  What we're doing here is waiting to do anything until we've handled every statement in the linked list of Blocks.

Reply:
Yeah, that is true. This node does get visited so that the linked list it represents can also be traversed.